src/content/blog/posts/github-one-stop-shop.md

```diff
@@ -0,0 +1,215 @@
+
+GitHub Discussions allows your users to [participate](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion) with the idea generation process through comments, reactions and polls. However, this means everyone involved will need to check their GitHub notifications regularly. It can also cause confusion if you already use another communication medium e.g. Slack. A good rule of thumb is to consider whether you might want to refer to this information in a year's time. If so, use a discussion!
+
+### Diagrams
+
+[Mermaid](https://github.com/mermaid-js/mermaid#readme) is a JavaScript-based tool that lets you create diagrams using a Markdown-like syntax. GitHub released support for Mermaid in [February 2022](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/). It's particularly handy for adding editable diagrams to places like issues and discussions, as you don't have to save it to the `/docs` folder. However I found that Mermaid has a steep learning curve and limited options available so I would only suggest for simple diagrams. 
+
+Instead, I recommend using [Excalidraw](https://excalidraw.com/), an open-source diagramming and whiteboarding tool. GitHub doesn't provide native support for Excalidraw, but you can export the diagram to SVG or PNG and embed the Excalidraw [scene data](https://kevinjalbert.com/using-excalidraws-embedded-scene-feature-for-collaborative-diagramming/) to make it editable. Whilst SVG is preferable because it is smaller in size and text-based (hence diffable), unfortunately it doesn't [render as nicely in GitHub](https://github.com/excalidraw/excalidraw/issues/4855) so I would stick with PNG for now. You can then upload the image file directly to issues and discussions, or save to a `/docs/images` folder for version control. GitHub has a nifty [image view mode](https://github.blog/2011-03-21-behold-image-view-modes/) for reviewing changes to images. You can edit diagrams on the Excalidraw web editor or download the [VS Code extension](https://marketplace.visualstudio.com/items?itemName=pomdtr.excalidraw-editor) to edit locally. You can also import diagrams directly from GitHub into the web editor by passing in the raw image url (see my Excalidraw diagram hyperlink as an example). Lastly, Excalidraw plus GitHub Discussions is great for recording the outcome of collaborative whiteboard sessions, which tend to get lost.
```

Comment:
```suggestion
Instead, I recommend using [Excalidraw](https://excalidraw.com/), an open-source diagramming and whiteboarding tool. GitHub doesn't provide native support for Excalidraw, but you can export the diagram to SVG or PNG and [embed the Excalidraw scene data](https://kevinjalbert.com/using-excalidraws-embedded-scene-feature-for-collaborative-diagramming/) to make it editable. Whilst SVG is preferable because it is smaller in size and is text-based (and hence diffable), unfortunately, it doesn't [render as nicely in GitHub](https://github.com/excalidraw/excalidraw/issues/4855) so I would stick with PNG for now. You can then upload the image file directly to issues and discussions, or save it to a `/docs/images` folder for version control. GitHub has a nifty [image view mode](https://github.blog/2011-03-21-behold-image-view-modes/) for reviewing changes to images. You can edit diagrams on the Excalidraw web editor or download the [VS Code extension](https://marketplace.visualstudio.com/items?itemName=pomdtr.excalidraw-editor) to edit locally. You can also import diagrams directly from GitHub into the web editor by passing in the raw image URL (see my Excalidraw diagram hyperlink as an example). Lastly, Excalidraw plus GitHub Discussions is great for recording the outcome of collaborative whiteboard sessions, which tend to get lost.
```